src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -266,18 +303,35 @@ public final void visitToken(DetailAST blockCommentNode) {
                 result = TREE_CACHE.get().get(treeCacheKey);
             }
             else {
-                result = context.get().parser.parseJavadocAsDetailNode(blockCommentNode);
+                final JavadocDetailNodeParser.ErrorListenerOption errorListenerOption;
+                if (violateExecutionOnNonTightHtml || !doesCheckAcceptJavadocWithNonTightHtml()) {
+                    errorListenerOption = JavadocDetailNodeParser
+                            .ErrorListenerOption.NOTIFY_FOR_NON_TIGHT_HTML;
+                }
+                else {
+                    errorListenerOption = JavadocDetailNodeParser.ErrorListenerOption.DEFAULT;
+                }
+                result = context.get().parser
+                        .parseJavadocAsDetailNode(blockCommentNode, errorListenerOption);
                 TREE_CACHE.get().put(treeCacheKey, result);
             }
 
             if (result.getParseErrorMessage() == null) {
                 processTree(result.getTree());
             }
             else {
-                final ParseErrorMessage parseErrorMessage = result.getParseErrorMessage();
-                log(parseErrorMessage.getLineNumber(),
-                        parseErrorMessage.getMessageKey(),
-                        parseErrorMessage.getMessageArguments());
+                final DetailNode javadocTree = result.getTree();
+
+                if (javadocTree != null && doesCheckAcceptJavadocWithNonTightHtml()) {
```

Comment:
@romani 
It might be so that I didn't properly understand what you meant but I don't understand why do you check `result.isTightHtml() && doesCheckAcceptJavadocWithNonTightHtml()`. Do you mean to say that we should check `!result.isTightHtml()` ? Because if the check accepts javadoc with non-tight html it seems unnecessary to check `result.isTightHtml`.

Also in the current design, `result.getParseErrorMessage() != null && result.getTree() != null` -> `javadoc tree has non tight HTML`